### PR TITLE
US-7.3: User Portfolio Page

### LINKED ISSUE #267: US-7.3: User Portfolio
**User Story Description**

As a user, I want to have a portfolio of all the events that I’ve participated in on my profile so that my friends and I are able to look back at them.

**User Acceptance Flow**

1. User logs in
1. User goes to their profile page
1. User clicks on 'My Portfolio'
1. User sees a portfolio of their past events

**Acceptance Criteria**

- [ ] 'My Portfolio' button redirects to user's portfolio page
- [ ] Join button (on event card and on event page) are not shown for portfolio events
- [ ] User Acceptance Flow is respected
- [ ] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [ ] Documentation present where necessary
- [ ] Unit and UI tests are implemented for every new component and function
- [ ] The application should still work with seeded data
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2047%3A75400)

**HINT**
- [Portfolio query already exists](https://github.com/RGPosadas/Mull/blob/develop/apps/mull-api/src/app/event/event.service.ts#L170)

### PR DESCRIPTION
**Pull request by JimLau49**
Closes #267 

**Pre-requisites**
Ensure that the user you are currently on has attended events that are now over in the DB

Steps to test: 
1. Run `mull-ui` and `mull-api`
2. Go to profile page
3. Click on My portfolio
4. Observe the portfolio events

### CODE REVIEW

File: apps/mull-ui/src/app/pages/profile/user-profile/user-profile.tsx [hunk 1]
```diff
@@ -40,7 +40,9 @@ export const UserProfilePage = ({ history }: UserProfilePageProps) => {
       />
       <div className="settings-container">
         <h3>Portfolio</h3>
-        <SettingsButton icon={faLeaf} text="My Portfolio" />
+        <Link to="/profile/portfolio">
```
Comment (cindyslittleplanet) [inline apps/mull-ui/src/app/pages/profile/user-profile/user-profile.tsx]:
```suggestion
        <Link to={ROUTES.PROFILE.PORTFOLIO}>
```

**Review by cindyslittleplanet — COMMENTED** [review]
Functionality works for me! 
However, I found a linting issue, a failing snapshot, and the E2E is failing for me - it can't find the event card since there isn't any, but E2Es are usually flaky on my side, so I need someone else to confirm it.

File: apps/mull-ui-e2e/src/integration/US-7.3.spec.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    it(`should show the user's portfolio`, () => {
+      cy.visit('http://localhost:4200/profile/portfolio');
+      cy.intercept('POST', 'http://localhost:3333/graphql', (req) => {
+        if (req.body.operationName === 'UserPortfolioEvents') {
+          req.reply({
+            statusCode: 200,
+            body: {
+              data: {
+                portfolioEvents: [
```
Comment (RGPosadas) [inline apps/mull-ui-e2e/src/integration/US-7.3.spec.ts]:
```suggestion
                userPortfolioEvents: [
```

**Review by RGPosadas — COMMENTED** [review]
Make sure to fix the linting error as well

**Review by RGPosadas — APPROVED** [review]
Thanks for addressing my comments, LGTM 🚀 

**Acceptance Criteria**

- [x] 'My Portfolio' button redirects to user's portfolio page
- [x] Join button (on event card and on event page) are not shown for portfolio events
- [x] User Acceptance Flow is respected
- [x] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2047%3A75400)

**Review by cindyslittleplanet — APPROVED** [review]
LGTM now! 🚀 

**Acceptance Criteria**

- [x] 'My Portfolio' button redirects to user's portfolio page
- [x] Join button (on event card and on event page) are not shown for portfolio events
- [x] User Acceptance Flow is respected
- [x] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2047%3A75400)

**Pull request merged**